main.go

```diff
@@ -91,7 +96,7 @@ func (p Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddyhttp.
 		return nil
 	}
 
-	to, err := p.authRedirect(w, r)
+	err := p.authRedirect(w, r)
```

Comment:
Could merge this and if statement below.

```suggestion
	if err := p.authRedirect(w, r); err != nil {
```